website/docs/commands/publishing/publish.md

```diff
@@ -0,0 +1,243 @@
+
+To override this default value, set the `publish.organization` directive, like
+```scala
+//> using publish.organization "io.github.foo"
+```
+
+### Name
+
+Scala CLI will use the project directory name as default Maven name. That is, if your
+Scala CLI project lives in a directory named `something`, it will be published as
+`something` (pure Java project) or `something_3` (Scala 3 project) for example.
+
+To override this default value, set the `publish.name` directive, like
+```scala
+//> using publish.name "something"
+```
+
+### Version
+
+If your Scala CLI project lives in a git repository, Scala CLI will infer a way to compute
+versions from it: if the current commit has a tag `v1.2.3`, version `1.2.3` is assumed.
+Else, if it has such a tag earlier in the git history, version `1.2.4-SNAPSHOT` is assumed.
+
+To override this default value, set the `publish.computeVersion` directive, like
+```scala
+//> using publish.computeVersion "git:tag"
+```
+
+## Repository settings
+
+A repository is required for the `publish` command, and might need other settings to work fine
+(to pass credentials for example).
+
+When publishing from you CI, we recommend letting `scala-cli publish setup`
+setting those settings via using directives. When publishing from your local machine to Maven Central,
+we recommend setting the repository via a `publish.repository` directive, and keeping your
+Sonatype credentials in the Scala CLI settings, via commands such as
+```bash
+SONATYPE_USER=… scala-cli config sonatype.user env:SONATYPE_USER --password-value
+SONATYPE_PASSWORD=… scala-cli config sonatype.password env:SONATYPE_PASSWORD
+```
+
+<!-- TODO Automatically generate that? -->
+|          | `using` directive | Command-line option | Example values | Notes |
+|----------|-------------------|---------------------|----------------|-------|
+| Repository | `publish.repository` | `--publish-repository` | `central`, `central-s01`, `github`, `https://artifacts.company.com/maven` | |
+| Repository User | `publish.user` | `--user` | `env:SONATYPE_USER` | Password value format |
+| Repository Password | `publish.password` | `--password` | `env:SONATYPE_PASSWORD` |
+| Repository Realm | `publish.realm` | `--realm` | | |
+
+## Other settings
+
+A number of metadata can be set either by `using` directives, or from the command-line. These
+metadata are optional in the `publish local` command, but might be mandatory for some repositories
+in the `publish` command, like Maven Central for non-snapshot versions.
+
+We recommend setting the settings below via using directives rather than on the command-line,
+so that these don't have to be recalled for each `scala-cli publish` or `scala-cli publish local`
+invocation. Command-line options for those settings take over the using directive values, and are
+provided as a convenience.
+
+<!-- TODO Automatically generate that? -->
+|          | `using` directive | Command-line option | Example values | Notes |
```

Comment:
I wonder if we shouldn't explain what options means, since Licences is self-descriptive but VSC may not be